runtime/include/cloe/databroker.hpp

```diff
@@ -0,0 +1,1254 @@
+
+    // Getter-function types
+    template <typename TValue>
+    using typed_get_value_function_t = std::function<const TValue*()>;
+    using type_erased_get_value_function_t = std::any;
+    // Setter-function types
+    template <typename TValue>
+    using typed_set_value_function_t = std::function<void(const TValue*)>;
+    using type_erased_set_value_function_t = std::any;
+    // Event types
+    template <typename TValue>
+    using typed_value_changed_event_t = Event<const TValue*>;
+    using type_erased_value_changed_event_t = std::any;    
+    // Event trigger-function types
+    template<typename TValue>
+    using typed_on_value_change_event_function_t = std::function<void(const TValue*)>;
+    using type_erased_on_value_change_event_function_t = std::any;
+
+    /**
+     * Name of the signal
+     */
+    std::vector<std::string> names_{};
+    /**
+     * std::type_info of the signal
+     */
+    const std::type_info* type_ {nullptr};
+    /**
+     * getter-function
+     */
+    type_erased_get_value_function_t get_value_{};
+    /**
+     * setter-function
+     */
+    type_erased_set_value_function_t set_value_{};
+    /**
+     * Event which gets raised when the signal value changes
+     */
+    type_erased_value_changed_event_t value_changed_event_{};
+    /**
+     * Triggers the value-changed event
+     */
+    type_erased_on_value_change_event_function_t on_value_changed_{};
+    /**
+     * std::function returning the count of event subscribers
+     */
+    std::function<std::size_t()> subscriber_count_{};
+
+    /**
+     * @brief Private default c'tor
+     * @note: Implementation-#2: The class shall be created only via a factory-method
+     */
+    Signal() = default;
+public:
+    /**
+     * @brief Public c'tor, accessible only via private access-token
+     * @note: Design-#1: The class shall be constructible via external helpers
+     */
+    explicit Signal(access_token):Signal(){}
+    Signal(const Signal&) = delete;
+    Signal(Signal&&) = default;
+    virtual ~Signal() = default;
+    Signal& operator =(const Signal&) = delete;
+    Signal& operator =(Signal&&) = default;
+
+private:
+    /**
+     * @brief Validates that the stored signal type matches the template type
+     * @tparam TValue Expected type of the signal
+     */
+    template<typename TValue>
+    void validate_type() const
+    {
+        const std::type_info* requested_type = &typeid(TValue);
+        if ((type_ == nullptr) || (requested_type == nullptr) || (type_ != requested_type))
+        {
+            throw std::logic_error("Type mismatch between signal and requested type");
+        }
+    }
+
+public:
+    /**
+     * @brief Returns the getter function of the signal
+     * @tparam TValue Type of the signal
+     * @return Getter function of the signal
+     */
+    template<typename TValue>
+    typename enable_if_is_wanted<TValue, const typed_get_value_function_t<TValue>*>::type getter() const

```

Comment:
It would be nice if this works:

```suggestion
    std::function<const TValue&()> getter() const
```

Reply:
As outlined, the design intent had two main aspects (in order of priority):
1) Prevent that error-prone data types are used.
2) Provide a meaningful error message, instead of typical C++ gibberish error messages.

ad 1) The potential need for references or pointers is virtually non-existent. The sole purpose of the databroker is to provide a kind of 'reflection mechanism' which breaks down the complexity of arbitrary data-types to intrinsics. Any more complex data-type would impose additional integration efforts elsewhere.
ad 2) UX.

Reply:
I suggest to re-group the code in a way that both groups, the functional and error-handling overloads, are grouped together. Additionally the error-handlers can be put into *.inc files.

Reply:
Utilizing a static_assert in the body the declaration looks like this

const typed_get_value_function_t<T>* getter() const